src/centengine/engine_impl.cc

```diff
@@ -66,6 +76,143 @@ grpc::Status engine_impl::GetStats(grpc::ServerContext* /*context*/,
   return grpc::Status::OK;
 }
 
+grpc::Status engine_impl::GetNbrHost(grpc::ServerContext* context, const ::google::protobuf::Empty* request, GenericValue* response) {
+  std::promise<int> p;
+  std::future<int> f1 = p.get_future();
+	
+  auto lambda = [&p]() -> int { 
+	p.set_value(host::hosts.size()); 
+	return 0;
+  }; 
+
+  command_manager::instance().enqueue(lambda);
+	
+  int val = f1.get();
+  response->set_value(val);	
+
+  return grpc::Status::OK;
+}
+
+grpc::Status engine_impl::GetNbrContact(grpc::ServerContext* context, const ::google::protobuf::Empty* request, GenericValue* response) {
+  std::promise<int> p;
+  std::future<int> f1 = p.get_future();
+  
+  auto lambda = [&p]() -> int {
+    p.set_value(contact::contacts.size()); 
+	return 0;
+  }; 
+
+  command_manager::instance().enqueue(lambda);
+	
```

Comment:
You can keep this as is but, why not this
command_manager::instance().enqueue([&p]() -> int {
  p.set_value...
  return 0;
});